src/middleware/auth.js

```diff
@@ -1,20 +1,32 @@
+
 const jwt = require('jsonwebtoken')
 const User = require('../models/User')
 
-const auth = async (req, res, next) => {
-    try {
-        const token = req.cookies.jwt
-        const decoded = jwt.verify(token, process.env.JWT_SECRET)
-        const user = await User.findOne({
-            _id: decoded._id,
-        })
-        if (!user) {
-            res.redirect('/login')
-        }
-        req.token = token //token
-        req.user = user
-    } catch (e) {
-        res.status(401).send({ error: 'Please authenticate.' })
-    }
-}
-module.exports = auth //auth to be exported
+
+require('dotenv').config();
+
+const requireAuth = (req, res, next) => {
+  const token = req.cookies.jwt;
+  console.log(token);
+  // check json web token exists & is verified
+  if (token) {
+    jwt.verify(token, process.env.JWT_SECRET, (err, decodedToken) => {
+      if (err) {
+        console.log(err.message);
+        
```

Comment:
Where is the req.user being populated?

Reply:
Asking the same thing - where is the req.user being populated and how is the token accessible on the frontend?